ocaml/src/osd_access.ml

```diff
@@ -40,33 +40,63 @@ class osd_access
     try Lwt.return (Hashtbl.find osds_info_cache osd_id)
     with Not_found ->
       mgr_access # get_osd_by_osd_id ~osd_id >>= fun osd_o ->
-      let osd_info = match osd_o with
-        | None -> failwith (Printf.sprintf "could not find osd with id %li" osd_id)
-        | Some info -> info
-      in
-      let osd_state = Osd_state.make () in
-      let info' = (osd_info, osd_state) in
-      Hashtbl.replace osds_info_cache osd_id info';
-      let long_id = let open Osd in
-                    Nsm_model.OsdInfo.get_long_id osd_info.kind
-      in
-      osd_long_id_claim_info :=
-        StringMap.add
-          long_id (Osd.ClaimInfo.ThisAlba osd_id)
-          !osd_long_id_claim_info;
-      Lwt.return info'
+      if not (Hashtbl.mem osds_info_cache osd_id)
+      then
+        begin
+          let osd_info = match osd_o with
+            | None -> failwith (Printf.sprintf "could not find osd with id %li" osd_id)
+            | Some info -> info
+          in
+          let osd_state = Osd_state.make () in
+          let info' = (osd_info, osd_state) in
+          Hashtbl.replace osds_info_cache osd_id info';
+          let long_id = let open Osd in
+                        Nsm_model.OsdInfo.get_long_id osd_info.kind
+          in
+          osd_long_id_claim_info :=
+            StringMap.add
+              long_id (Osd.ClaimInfo.ThisAlba osd_id)
+              !osd_long_id_claim_info;
+
+          Lwt_extra2.run_forever
+            "refresh_osd_total_used"
+            (fun () ->
+             Pool.Osd.factory
+               osd_buffer_pool
+               (Hashtbl.find osds_info_cache osd_id |> fst).Osd.kind
+             >>= fun (osd, closer) ->
+             Lwt.finalize
+               (fun () ->
+                osd # get_disk_usage >>= fun (used, total) ->
+                Osd_state.add_used_total osd_state used total;
```

Comment:
everywhere else, it's `_disk_usage` and here it's _used_total... It would make sense to use the same naming scheme. Also, the argument could be a tuple.